src/profiler.cpp

```diff
@@ -1604,11 +1620,16 @@ void Profiler::dumpText(Writer& out, Arguments& args) {
             out << buf;
 
             CallTrace* trace = it->trace;
-            for (int j = 0; j < trace->num_frames; j++) {
+            int j = 0;
+            for (; j < trace->num_frames; j++) {
                 const char* frame_name = fn.name(trace->frames[j]);
                 snprintf(buf, sizeof(buf) - 1, "  [%2d] %s\n", j, frame_name);
                 out << buf;
             }
+            if (trace->truncated) {
+                snprintf(buf, sizeof(buf) - 1, "  [%2d] [truncated]\n", j);
```

Comment:
All truncated stack traces obtained with cstack=vm/vmx should ideally end with an error frame.
These are `[break_compiled]`, `[break_interpreted]` etc. in the current implementation, but we can change names if needed. If an error frame is missed in some cases, let's fix it.

Reply:
This may not be the case if you set the maximum stackdepth - as I do in the tests. Without outputting the synthetic root frame you have no idea that a stacktrace is not complete.